server/src/main/java/org/opensearch/cluster/routing/WRRShardsCache.java

```diff
@@ -0,0 +1,95 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.cluster.routing;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.opensearch.cluster.ClusterChangedEvent;
+import org.opensearch.cluster.ClusterStateListener;
+import org.opensearch.cluster.service.ClusterService;
+import org.opensearch.common.cache.Cache;
+import org.opensearch.common.cache.CacheBuilder;
+import org.opensearch.common.lease.Releasable;
+import org.opensearch.index.shard.ShardId;
+
+import java.util.List;
+
+/**
+ * The WRR shards cache allows caching shard ordering returned by Weighted round-robin scheduling policy ,helping with
+ * improving similar requests.
+ *
+ * @opensearch.internal
+ */
+public class WRRShardsCache implements Releasable, ClusterStateListener {
+    private static final Logger logger = LogManager.getLogger(WRRShardsCache.class);
+
+    private final Cache<Key, List<ShardRouting>> cache;
+
+    public WRRShardsCache(ClusterService clusterService) {
+
+        final long sizeInBytes = 2000000;
```

Comment:
Make this static final. Also how did we arrive at this size?

Reply:
An average shard routing info size in cache is 75 bytes, so if there are 3 shard copies around 8k shards info can be stored in cache if we limit cache size to 2MB. But looks like until the weigher function is defined, cache consider size as 1 byte for a record irrespective of its size. Any pointers on how to calculate object size in weigher function?

Reply:
removed WRRShardsCache